src/SSVOpenHexagon/Core/HGGraphics.cpp

```diff
@@ -175,20 +191,34 @@ void HexagonGame::draw()
             playerTris3D.unsafe_emplace_other(playerTris);
         }
 
+        if(renderAbove)
+        {
+            for(std::size_t i = 0; i < aboveMain; ++i)
+            {
+                wallQuads3DTop.unsafe_emplace_other(wallQuads);
+                pivotQuads3DTop.unsafe_emplace_other(pivotQuads);
+                playerTris3DTop.unsafe_emplace_other(playerTris);
+            }
+        }
+
         const auto adjustAlpha = [&](sf::Color& c, const float i)
         {
             SSVOH_ASSERT(styleData._3dAlphaMult != 0.f);
 
             const float newAlpha =
                 (static_cast<float>(c.a) / styleData._3dAlphaMult) -
-                i * styleData._3dAlphaFalloff;
+                (styleData._3dAlphaMirror ? std::abs(i + 1) : i) *
+                    styleData._3dAlphaFalloff;
```

Comment:
Ditto, this expression is pretty complicated. Either comments or maybe splitting it up would help.